From ae934815098635db4b42522dcc401ed2c8d7f991 Mon Sep 17 00:00:00 2001
From: "Nguyen Van Linh [FGA.BU13]" <LinhNV25@fsoft.com.vn>
Date: Mon, 9 Apr 2018 16:26:05 +0700
Subject: [PATCH 202/642] ARM: dts: r8a77470: Add clocks

Declare all core clocks and DIV6 clocks, as well as all MSTP clocks
currently used by r8a77470 boards.

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a77470.dtsi | 343 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 343 insertions(+)

diff --git a/arch/arm/boot/dts/r8a77470.dtsi b/arch/arm/boot/dts/r8a77470.dtsi
index 83866b2..2f093948 100644
--- a/arch/arm/boot/dts/r8a77470.dtsi
+++ b/arch/arm/boot/dts/r8a77470.dtsi
@@ -8,6 +8,7 @@
  * kind, whether express or implied.
  */
 
+#include <dt-bindings/clock/r8a77470-clock.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 
@@ -52,4 +53,346 @@
 			<0 0xf1006000 0 0x2000>;
 		interrupts = <1 9 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_HIGH)>;
 	};
+
+	clocks {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* External root clock */
+		extal_clk: extal_clk {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			/* This value must be overriden by the board. */
+			clock-frequency = <0>;
+			clock-output-names = "extal";
+		};
+
+		/*
+		 * The external audio clocks are configured as 0 Hz fixed frequency clocks by
+		 * default. Boards that provide audio clocks should override them.
+		 */
+		audio_clk_a: audio_clk_a {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <0>;
+			clock-output-names = "audio_clk_a";
+		};
+		audio_clk_b: audio_clk_b {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <0>;
+			clock-output-names = "audio_clk_b";
+		};
+		audio_clk_c: audio_clk_c {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <0>;
+			clock-output-names = "audio_clk_c";
+		};
+
+		/* Special CPG clocks */
+		cpg_clocks: cpg_clocks@e6150000 {
+			compatible = "renesas,r8a77470-cpg-clocks",
+				     "renesas,rcar-gen2-cpg-clocks";
+			reg = <0 0xe6150000 0 0x1000>;
+			clocks = <&extal_clk>;
+			#clock-cells = <1>;
+			clock-output-names = "main", "pll0", "pll1", "pll3",
+					     "lb", "qspi", "sdh", "sd0", "sd1", "z";
+		};
+
+		/* Variable factor clocks */
+		sd2_clk: sd2_clk@e6150078 {
+			compatible = "renesas,r8a77470-div6-clock", "renesas,cpg-div6-clock";
+			reg = <0 0xe6150078 0 4>;
+			clocks = <&pll1_div2_clk>;
+			#clock-cells = <0>;
+			clock-output-names = "sd2";
+		};
+
+		/* Fixed factor clocks */
+		pll1_div2_clk: pll1_div2_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <2>;
+			clock-mult = <1>;
+			clock-output-names = "pll1_div2";
+		};
+		z2_clk: z2_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL0>;
+			#clock-cells = <0>;
+			clock-div = <1>;
+			clock-mult = <1>;
+			clock-output-names = "z2";
+		};
+		zx_clk: zx_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <3>;
+			clock-mult = <1>;
+			clock-output-names = "zx";
+		};
+		zs_clk: zs_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <6>;
+			clock-mult = <1>;
+			clock-output-names = "zs";
+		};
+		hp_clk: hp_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <12>;
+			clock-mult = <1>;
+			clock-output-names = "hp";
+		};
+		b_clk: b_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <12>;
+			clock-mult = <1>;
+			clock-output-names = "b";
+		};
+		p_clk: p_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <24>;
+			clock-mult = <1>;
+			clock-output-names = "p";
+		};
+		cl_clk: cl_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <48>;
+			clock-mult = <1>;
+			clock-output-names = "cl";
+		};
+		m2_clk: m2_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <8>;
+			clock-mult = <1>;
+			clock-output-names = "m2";
+		};
+		rclk_clk: rclk_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <(48 * 1024)>;
+			clock-mult = <1>;
+			clock-output-names = "rclk";
+		};
+		oscclk_clk: oscclk_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <(12 * 1024)>;
+			clock-mult = <1>;
+			clock-output-names = "oscclk";
+		};
+		zb3_clk: zb3_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL3>;
+			#clock-cells = <0>;
+			clock-div = <8>;
+			clock-mult = <1>;
+			clock-output-names = "zb3";
+		};
+		mp_clk: mp_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&pll1_div2_clk>;
+			#clock-cells = <0>;
+			clock-div = <15>;
+			clock-mult = <1>;
+			clock-output-names = "mp";
+		};
+		cp_clk: cp_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&cpg_clocks R8A77470_CLK_PLL1>;
+			#clock-cells = <0>;
+			clock-div = <48>;
+			clock-mult = <1>;
+			clock-output-names = "cp";
+		};
+
+		acp_clk: acp_clk {
+			compatible = "fixed-factor-clock";
+			clocks = <&extal_clk>;
+			#clock-cells = <0>;
+			clock-div = <2>;
+			clock-mult = <1>;
+			clock-output-names = "acp";
+		};
+
+		/* Gate clocks */
+		mstp0_clks: mstp0_clks@e6150130 {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150130 0 4>, <0 0xe6150030 0 4>;
+			clocks = <&mp_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <R8A77470_CLK_MSIOF0>;
+			clock-output-names = "msiof0";
+		};
+		mstp1_clks: mstp1_clks@e6150134 {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150134 0 4>, <0 0xe6150038 0 4>;
+			clocks = <&zs_clk>, <&zs_clk>, <&p_clk>, <&zs_clk>, <&zs_clk>,
+				 <&zs_clk>, <&p_clk>, <&p_clk>, <&rclk_clk>,
+				 <&zs_clk>, <&zs_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_VCP0 R8A77470_CLK_VPC0 R8A77470_CLK_TMU1
+				R8A77470_CLK_PVRSRVKM R8A77470_CLK_2DDMAC R8A77470_CLK_FDP0
+				R8A77470_CLK_TMU3 R8A77470_CLK_TMU2 R8A77470_CLK_CMT0
+				R8A77470_CLK_VSP1_DU0 R8A77470_CLK_VSP1_S
+			>;
+			clock-output-names =
+				"vcp0", "vpc0", "tmu1", "pvrsrvkm", "tddmac", "fdp0",
+				"tmu3", "tmu2", "cmt0", "vsp1-du0", "vsps";
+		};
+		mstp2_clks: mstp2_clks@e6150138 {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150138 0 4>, <0 0xe6150040 0 4>;
+			clocks = <&mp_clk>, <&mp_clk>,
+				 <&zs_clk>, <&zs_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_MSIOF2 R8A77470_CLK_MSIOF1
+				R8A77470_CLK_SYS_DMAC1 R8A77470_CLK_SYS_DMAC0
+			>;
+			clock-output-names =
+				"msiof2", "msiof1",
+				"sys-dmac1", "sys-dmac0";
+		};
+		mstp3_clks: mstp3_clks@e615013c {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe615013c 0 4>, <0 0xe6150048 0 4>;
+			clocks = <&sd2_clk>, <&cpg_clocks R8A77470_CLK_SD1>, <&cpg_clocks R8A77470_CLK_SD0>,
+				 <&hp_clk>, <&hp_clk>,
+				 <&rclk_clk>, <&hp_clk>, <&hp_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_SDHI2 R8A77470_CLK_SDHI1 R8A77470_CLK_SDHI0
+				R8A77470_CLK_USBDMAC0_1 R8A77470_CLK_USBDMAC1_1
+				R8A77470_CLK_CMT1 R8A77470_CLK_USBDMAC0_0 R8A77470_CLK_USBDMAC1_0
+			>;
+			clock-output-names =
+				"sdhi2", "sdhi1", "sdhi0",
+				"usbdmac0_1", "usbdmac1_1",
+				"cmt1", "usbdmac0_0", "usbdmac1_0";
+		};
+		mstp5_clks: mstp5_clks@e6150144 {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150144 0 4>, <0 0xe615003c 0 4>;
+			clocks = <&zs_clk>, <&p_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <R8A77470_CLK_AUDIO_DMAC0 R8A77470_CLK_PWM>;
+			clock-indices = <R8A77470_CLK_AUDIO_DMAC0 R8A77470_CLK_PWM>;
+			clock-output-names = "audmac0", "pwm";
+		};
+		mstp7_clks: mstp7_clks@e615014c {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe615014c 0 4>, <0 0xe61501c4 0 4>;
+			clocks = <&mp_clk>, <&hp_clk>, <&mp_clk>, <&hp_clk>, <&zs_clk>,
+				 <&p_clk>, <&p_clk>, <&zs_clk>, <&zs_clk>, <&p_clk>, <&p_clk>,
+				 <&p_clk>, <&p_clk>, <&zx_clk>, <&zx_clk>, <&zx_clk>, <&zx_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_EHCI0 R8A77470_CLK_HSUSB0 R8A77470_CLK_EHCI1
+				R8A77470_CLK_HSUSB1 R8A77470_CLK_HSCIF2
+				R8A77470_CLK_SCIF5 R8A77470_CLK_SCIF4 R8A77470_CLK_HSCIF1
+				R8A77470_CLK_HSCIF0 R8A77470_CLK_SCIF3 R8A77470_CLK_SCIF2
+				R8A77470_CLK_SCIF1 R8A77470_CLK_SCIF0 R8A77470_CLK_DU1
+				R8A77470_CLK_DU0 R8A77470_CLK_LVDS0 R8A77470_CLK_DVENC
+			>;
+			clock-output-names =
+				"ehci0", "hsusb0", "ehci1", "hsusb1", "hscif2",
+				"scif5", "scif4", "hscif1", "hscif0", "scif3",
+				"scif2", "scif1", "scif0", "du1", "du0", "lvds0", "dvenc";
+		};
+		mstp8_clks: mstp8_clks@e6150990 {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150990 0 4>, <0 0xe61509a0 0 4>;
+			clocks = <&zx_clk>, <&zx_clk>, <&zs_clk>, <&zs_clk>, <&hp_clk>, <&p_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_IPMMU_GP R8A77470_CLK_DVDEC
+				R8A77470_CLK_VIN1 R8A77470_CLK_VIN0 R8A77470_CLK_AVB
+				R8A77470_CLK_ETHER
+			>;
+			clock-output-names =
+				"ipmmu_gp", "dvdec", "vin1", "vin0", "avb", "ether";
+		};
+		mstp9_clks: mstp9_clks@e6150994 {
+			compatible = "renesas,r8a77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150994 0 4>, <0 0xe61509a4 0 4>;
+			clocks = <&cp_clk>, <&cp_clk>, <&cp_clk>,
+				 <&cp_clk>, <&cp_clk>, <&cp_clk>,
+				 <&p_clk>, <&p_clk>,
+				 <&cpg_clocks R8A77470_CLK_QSPI>, <&cpg_clocks R8A77470_CLK_QSPI>,
+				 <&hp_clk>, <&hp_clk>, <&hp_clk>,
+				 <&hp_clk>, <&hp_clk>;
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_GPIO5 R8A77470_CLK_GPIO4 R8A77470_CLK_GPIO3
+				R8A77470_CLK_GPIO2 R8A77470_CLK_GPIO1 R8A77470_CLK_GPIO0
+				R8A77470_CLK_RCAN0 R8A77470_CLK_RCAN1
+				R8A77470_CLK_QSPI_MOD1 R8A77470_CLK_QSPI_MOD0
+				R8A77470_CLK_I2C4 R8A77470_CLK_I2C3 R8A77470_CLK_I2C2
+				R8A77470_CLK_I2C1 R8A77470_CLK_I2C0
+			>;
+			clock-output-names =
+				"gpio5", "gpio4", "gpio3", "gpio2", "gpio1", "gpio0", "rcan0", "rcan1",
+				"qspi_mod1", "qspi_mod0", "i2c4", "i2c3", "i2c2", "i2c1", "i2c0";
+		};
+		mstp10_clks: mstp10_clks@e6150998 {
+			compatible = "renesas,R8A77470-mstp-clocks", "renesas,cpg-mstp-clocks";
+			reg = <0 0xe6150998 0 4>, <0 0xe61509a8 0 4>;
+			clocks = <&p_clk>, /* parent of SCU */
+				<&p_clk>,
+				<&p_clk>, <&p_clk>, <&p_clk>, <&p_clk>, <&p_clk>,
+				<&p_clk>, <&p_clk>, <&p_clk>, <&p_clk>, <&p_clk>,
+				<&mstp10_clks R8A77470_CLK_SCU>, <&mstp10_clks R8A77470_CLK_SCU>,
+				<&mstp10_clks R8A77470_CLK_SCU>, <&mstp10_clks R8A77470_CLK_SCU>,
+				<&mstp10_clks R8A77470_CLK_SCU>, <&mstp10_clks R8A77470_CLK_SCU>,
+				<&mstp10_clks R8A77470_CLK_SCU>, <&mstp10_clks R8A77470_CLK_SCU>;
+
+			#clock-cells = <1>;
+			renesas,clock-indices = <
+				R8A77470_CLK_SCU
+				R8A77470_CLK_SSI
+				R8A77470_CLK_SSI9 R8A77470_CLK_SSI8 R8A77470_CLK_SSI7 R8A77470_CLK_SSI6 R8A77470_CLK_SSI5
+				R8A77470_CLK_SSI4 R8A77470_CLK_SSI3 R8A77470_CLK_SSI2 R8A77470_CLK_SSI1 R8A77470_CLK_SSI0
+				R8A77470_CLK_DVC1 R8A77470_CLK_DVC0
+				R8A77470_CLK_SRC6 R8A77470_CLK_SRC5
+				R8A77470_CLK_SRC4 R8A77470_CLK_SRC3 R8A77470_CLK_SRC2 R8A77470_CLK_SRC1
+			>;
+			clock-indices = <
+				R8A77470_CLK_SCU
+				R8A77470_CLK_SSI
+				R8A77470_CLK_SSI9 R8A77470_CLK_SSI8 R8A77470_CLK_SSI7 R8A77470_CLK_SSI6 R8A77470_CLK_SSI5
+				R8A77470_CLK_SSI4 R8A77470_CLK_SSI3 R8A77470_CLK_SSI2 R8A77470_CLK_SSI1 R8A77470_CLK_SSI0
+				R8A77470_CLK_DVC1 R8A77470_CLK_DVC0
+				R8A77470_CLK_SRC6 R8A77470_CLK_SRC5
+				R8A77470_CLK_SRC4 R8A77470_CLK_SRC3 R8A77470_CLK_SRC2 R8A77470_CLK_SRC1
+			>;
+			clock-output-names =
+				"scu", "ssi",
+				"ssi9", "ssi8", "ssi7", "ssi6", "ssi5",
+				"ssi4", "ssi3", "ssi2", "ssi1", "ssi0",
+				"dvc1", "dvc0",
+				"src6", "src5",
+				"src4", "src3", "src2", "src1";
+		};
+	};
 };
\ No newline at end of file
-- 
2.7.4

